tests/interactive/test_rewrites.py

```diff
@@ -0,0 +1,120 @@
+from xdsl.dialects.builtin import (
+    StringAttr,
+)
+from xdsl.dialects.test import TestOp
+from xdsl.interactive.get_condensed_passes import AvailablePass
+from xdsl.interactive.rewrites import (
+    IndexedIndividualRewrite,
+    IndividualRewrite,
+    convert_indexed_individual_rewrites_to_available_pass,
+    get_all_possible_rewrites,
+)
+from xdsl.ir import MLContext
+from xdsl.parser import Parser
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.tools.command_line_tool import get_all_dialects
+from xdsl.transforms import individual_rewrite
+from xdsl.utils.parse_pipeline import parse_pipeline
+
+
+class Rewrite(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: TestOp, rewriter: PatternRewriter):
+        if op.attributes["label"] != StringAttr("a"):
+            return
+        rewriter.replace_matched_op(TestOp(attributes={"label": StringAttr("c")}))
+
+
+def test_get_all_possible_rewrite():
+    # build module
+    prog = """
+    builtin.module {
+    "test.op"() {"label" = "a"} : () -> ()
+    "test.op"() {"label" = "a"} : () -> ()
+    "test.op"() {"label" = "b"} : () -> ()
+    }
+    """
+
+    ctx = MLContext(True)
+    for dialect_name, dialect_factory in get_all_dialects().items():
+        ctx.register_dialect(dialect_name, dialect_factory)
+    parser = Parser(ctx, prog)
+    module = parser.parse_module()
+
+    expected_res = (
+        (
+            IndexedIndividualRewrite(
+                1, IndividualRewrite(operation="test.op", pattern="TestRewrite")
+            )
+        ),
+        (
+            IndexedIndividualRewrite(
+                operation_index=2,
+                rewrite=IndividualRewrite(operation="test.op", pattern="TestRewrite"),
+            )
+        ),
+    )
+
+    res = get_all_possible_rewrites(module, {"test.op": {"TestRewrite": Rewrite()}})
+    assert res == expected_res
+
+
```

Comment:
everything after line 66 is new, everything before was in a previously named file test_get_all_possible_rewrites(). I renamed the file to test_rewrites().